src/change_stream.ts

```diff
@@ -235,19 +244,21 @@ export class ChangeStream extends EventEmitter {
     // Create contained Change Stream cursor
     this.cursor = createChangeStreamCursor(this, options);
 
+    const cursorStream = this.cursor[kCursorStream];
+    if (!cursorStream) throw new MongoError('No stream on cursor');
```

Comment:
This is confusing to me, this is the constructor of the `ChangeStream` and we're effectively asserting that the `cursorStream` will _always_ exist, which shouldn't be the case. If I do something like this:
```
const changeStream = collection.watch();
for await (const doc of changeStream) {
  console.dir({ doc });
}
```
A `cursorStream` should never have been created. In fact, you might want to add a test for that? 

I think what you want is to modify the `newListener` handler below. If someone registers a listener for `change`, then we want to set up a new `cursorStream` and forward the `data` event to them. Otherwise, if they want to stream from this change stream, they should call the `stream` method.

Reply:
That makes sense, I was also a bit confused why this ended up working and surprised no tests failed. I'll try the `newListener` approach out.

Reply:
@mbroadst I think I got this ironed out, PTAL 🙏